### PR TITLE
Django 1.9 compatibility issues

### PR DESCRIPTION
**Pull request by dracos**
Adds 1.9-1.11 support, makes tests actually run in python 3 (rather than python 2!), Travis container build, selenium switch to geckodriver.

### CODE REVIEW

**Comment by wfdd** [comment]
The installation instructions at https://mysociety.github.io/sayit/install/ will also need updating, as the `syncdb` command has been removed.

File: setup.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def read_file(filename):
         'pytz >= 2013d',
         'six >= 1.4.1',
         django,
-        'Django-Select2 == 4.3.2',
+        'mysociety-Django-Select2 == 4.3.2.1',
```
Comment (mhl) [inline setup.py]: Maybe this should be in a separate commit with an explanation of why we're using our own fork?

File: tox.ini [hunk 2]
```diff
@@ -1,16 +1,17 @@
 [tox]
-envlist = flake8, py-{27,34}-{1.8,1.9}
+envlist = flake8, py-{27,34}-{1.8,1.9,1.10}
 
 [testenv]
 commands =
     flake8: flake8
     py: pip install -e .[test]
-    py: python manage.py test
+    py: python manage.py test {posargs}
```
Comment (mhl) [inline tox.ini]: Why is this needed now, but not before?

File: speeches/tests/section_tests.py [hunk 3]
```diff
@@ -176,7 +176,7 @@ def test_add_section_with_heading(self):
             'heading': 'A test section'
         })
         new_section = Section.objects.order_by('-id')[0]
-        self.assertRedirects(resp, '%s' % new_section.slug)
+        self.assertRedirects(resp, '/%s' % new_section.slug)
```
Comment (dracos) [inline speeches/tests/section_tests.py]: Before 1.9 it added scheme+host (and presumably therefore the initial `/`) implicitly. All redirect checks are relative since 1.9.
